db/table_properties_collector.h

```diff
@@ -108,4 +110,60 @@ class UserKeyTablePropertiesCollectorFactory
   std::shared_ptr<TablePropertiesCollectorFactory> user_collector_factory_;
 };
 
+class TimestampTablePropertiesCollector : public IntTblPropCollector {
+ public:
+  explicit TimestampTablePropertiesCollector(const Comparator* cmp)
+      : cmp_(cmp) {}
+
+  virtual ~TimestampTablePropertiesCollector() {}
+
+  Status InternalAdd(const Slice& key, const Slice& /* value */,
+                     uint64_t /* file_size */) override {
+    if (!cmp_) {
+      return Status::OK();
+    }
+    auto timestampInKey =
+        ExtractTimestampFromUserKey(key, cmp_->timestamp_size());
+    if (timestampInKey == kDisableUserTimestamp) {
+      return Status::OK();
+    }
```

Comment:
Didn't check thoroughly, but can `cmp_` be null here?

Can this be
```
assert (cmp_);
if (cmp_->timestamp_size() == 0) {
  return Status::OK();
}
if (key.size() < cmp_->timestamp_size()) {
  return some error;
}
```

Reply:
fine. Already fixed it.

Reply:
Since now adding TimestampTablePropertiesCollector only when timestamp enabled, here assert change to

assert (cmp_ && cmp_->timestamp_size() > 0);